File: adminportal/src/Dto/AdminPortal/TopicInfo.ts
import type { ThreadBasicInfo } from "./ThreadInfo"
import type { UserBasicInfo, UserFullInfo } from "./UserInfo"

export type TopicEntry = {
    topicId: string,
    boardId: string,
    topicName: string,
    description: string,
    createdByUserId: string,
    createdDate: Date
}

export type TopicBasicInfo = {
    topic: TopicEntry,
    totalThreads: number,
    totalPosts: number,
    numNewthreads: number,
    newestThread: ThreadBasicInfo,
    createdByUser: UserBasicInfo
}

export type TopicFullInfo = {
    topic: TopicEntry,
    totalThreads: number,
    totalPosts: number,
    numNewthreads: number,
    newestThread: ThreadBasicInfo,
    createdByUser: UserBasicInfo,
    threads: ThreadBasicInfo[]
}

export type TopicSummary = {
    totalTopics: number
}

File: server/RestApiServer.Endpoints/Dto/Forum/TopicInfo.cs
using RestApiServer.Db;
using RestApiServer.Dto.App;

namespace RestApiServer.Dto.Forum
{
    public class TopicBasicInfo
    {
        public required TopicEntry Topic { get; set; }
        public int TotalThreads { get; set; } = 0;
        public int TotalPosts { get; set; } = 0;
        public int NumNewThreads { get; set; }
        public ThreadBasicInfo NewestThread { get; set; } = null!;
        public UserBasicInfo CreatedByUser { get; set; } = null!;
    }

    public class TopicFullInfo
    {
        public required TopicEntry Topic { get; set; }
        public required int TotalThreads { get; set; } = 0;
        public required int TotalPosts { get; set; } = 0;
        public int NumNewThreads { get; set; }
        public ThreadBasicInfo NewestThread { get; set; } = null!;
        public required UserBasicInfo CreatedByUser { get; set; }
        public List<ThreadBasicInfo> Threads { get; set; } = new();
    }
    public class TopicSummary 
    {
        public required int TotalTopics { get; set; } = 0;
    }
}